### DISCUSSION TITLE
Interactive time series

### DISCUSSION THREAD

**Discussion by renefritze** [marcomusy/vedo#493] [Q&A]
I'm trying to gauge the feasibility of using vedo for showing interactive time series visualization (of meshes) in both a Jupyter and Qt-Application embedded context.

AFAICT, there is no support/example for (prerecorded) animations in notebooks atm. Is that correct? If yes, is that on the horizon?

For the interactivity (play/pause/dragging a slider for the time steps), it seems that controls do not yet translate into the Jupyter environment?

**Comment by marcomusy** [comment]
Hi Rene, for qt in principle it should be possible to embed sliders and buttons like in the native rendering window (I will try myself and let you know).
For the Jupyter environment this is not possible, but you can still use the standard callbacks in [ipywidgets](https://ipywidgets.readthedocs.io/en/latest/examples/Widget%20Events.html) to interact with the scene.

**Reply by renefritze** [reply]
Ipywidgets are problematic for us, if the scene manipulation requires an active kernel. We're aiming for a fully interactive HTML/js representation for embedding in documentation/tutorial documents, if possible. 
Since you're (optionally) passing the rendering to K3D, this might be possible with vedo too, considering [this example](https://k3d-jupyter.org/showcase/nbody.html) has time+loop controls in the JS frontend menu thingy.

For the Qt part, I'd be happy to try this myself  now, since you consider it principally viable.
